### DISCUSSION TITLE
Autoposting

### DISCUSSION THREAD

**Discussion by jmduke** [buttondown/discussions#7] [Ideas]
## TL;DR

Connect your social network; Buttondown will automatically post new emails that are sent out.

## What's the problem?

Almost _everyone_, even me, has the same workflow:

1. Publish a post, in one way or another.
2. Spread the word about the post on a bunch of other channels.

That second item is manual when it really doesn't need to be. This is not a big deal for _manual_ publishers (who actually hit the 'send' button in Buttondown) but is particularly gnarly for folks who schedule their emails or trigger them via the API / RSS.

## Why implement it?

Unlike [surveys](https://github.com/buttondown-email/discussions/discussions/6), this work _is_ easily replicable by tools like Zapier. Why build it in-house?

Two reasons:

1. From a _technical_ perspective, the majority of the effort here comes from actually building out the OAuth support with various social networks and keeping tokens up to date — the actual 'autoposting' part is relatively trivial. As it so happens, I want to build out those OAuth pieces for an unrelated but more important project (social signin!), which significantly ameliorates the technical investment here.
4. I wrote in the product brief for [autoposting](https://github.com/buttondown-email/discussions/discussions/5) that I really like the idea of Buttondown being a 'headless newsletter' solution. This is very much in service of that.

## What's the solution?

The solution space here is pretty well-paved (see below), but more to come as I ideate further.

![image](https://user-images.githubusercontent.com/461046/215287974-e608e8b6-f865-432d-a6db-3e1c9f09fd32.png)

## Open questions

- [ ] What social networks to support? My initial guess is Twitter, Mastodon, and LinkedIn.

## Out of scope

🤔 Not sure, though I suppose that's largely a failure of imagination.

**Comment by greggcorp** [comment]
+1 to that cohort of social networks to prioritize.

**Comment by JayCuthrell** [comment]
I was looking for LinkedIn references in this org and I'm excited to learn more about the new [cross-post to LinkedIn](https://buttondown.email/settings/integrations) feature 🤓

**Reply by jmduke** [reply]
I would be lying if I said I didn't forget about this public post (though of course the link you dropped is a direct successor to this branch that got started 18 months ago!)

**Reply by JayCuthrell** [reply]
_Update:_ I've been attempting to use the Automation for LinkedIn, Bluesky, and an email template notification but I may not understand email variables.

Is there a working example that passed a LinkedIn test with two (2) email variables?

So far I can make one email variable work but not two (2) email variables on LinkedIn and Bluesky.

Example:

✅ Bluesky and LinkedIn `I've just posted {{email.subject}}`

✅ Bluesky and LinkedIn `Read more... {{email_url}}`

✅ Bluesky `I've just posted {{email.subject}}` 
        `    Read more... {{email_url}} `

🙅 LinkedIn `I've just posted {{email.subject}}` 
        `    Read more... {{email_url}} `

Possibly related, I can't make any Automation email template notifications with email variables as the email arrives without processing `{{email_url}}` and prints out the string "{{email_url}}"

**Reply by jmduke** [reply]
Hmmm, that's odd — I wonder if this is a LinkedIn thing about supplying URLs in a weird way? (We have the same rendering pipeline for both.)

Is this a fair and representative sample of the autopost you were/are attempting?:

```
And now... 🤔 💡 🤯 🤓

🌶️ Hot Fudge Sunday for 2025-04-20

https://hot.fudge.org/archive/hot-fudge-sunday-for-2025-04-20/```

**Reply by jmduke** [reply]
Oh and filing/fixing this!:

> Possibly related, I can't make any Automation email template notifications with email variables as the email arrives without processing {{email_url}} and prints out the string "{{email_url}}"

**Reply by JayCuthrell** [reply]
> Hmmm, that's odd — I wonder if this is a LinkedIn thing about supplying URLs in a weird way? (We have the same rendering pipeline for both.)
> 
> Is this a fair and representative sample of the autopost you were/are attempting?:
> 
> ```
> And now... 🤔 💡 🤯 🤓
> 
> 🌶️ Hot Fudge Sunday for 2025-04-20
> 
> https://hot.fudge.org/archive/hot-fudge-sunday-for-2025-04-20/```
> ```

Correct. It is functional and does share the URL but for whatever reason the LinkedIn post does not appear to be processed as an unfurl / rich media card as would be expected from sharing a URL. 

<img width="734" alt="Screenshot 2025-04-21 at 9 06 07 PM" src="https://github.com/user-attachments/assets/1c1642c0-675d-4076-bd4f-305d78835f76" />


I suspect this is due to not passing the opengraph image as would happen with a manually created post sharing a URL.

**Reply by jmduke** [reply]
Aha, yes — we need to technically _upload the image/link as a separate asset_. Thanks for the flag; we'll get this done soon.
